eng/docgeneration/Generate-DocIndex.ps1

```diff
@@ -12,142 +22,80 @@ Write-Verbose "Initializing Default DocFx Site..."
     # after processing the list of subdirectories. This will allow the correct 
     # division of the artifacts under the Client or Management headings
     $clientArr = @()
-    $mgmtArr = @()
-
-    $PkgList = Get-ChildItem $Dir.FullName -Directory -Exclude changelog, faq, .github, build
-    if (($PkgList | Measure-Object).count -eq 0)
-    {
-        continue
-    }
-    foreach ($Pkg in $PkgList)
-    {
-        # Load the pom file to pull the artifact name and grab the
-        # parent's relative path to see which parent pom is being
-        # used to determine whether or not the artifact is client 
-        # or management.
-        $PomPath = Join-Path -Path $Pkg -ChildPath "pom.xml"
-
-        # no pom file = nothing to process
-        if (Test-Path -path $PomPath) 
-        {
-            $xml = New-Object xml
-            $xml.Load($PomPath)
-
-            # Get the artifactId from the POM
-            $artifactId = $xml.project.artifactId
-
-            $parent = $xml.project.parent.relativePath
-
-            # If this is an artifact that isn't shipping then just
-            # move on to the next one
-            if ($ArtifactsToSkip -contains $artifactId)
-            {
-                Write-Output "skipping $artifactId"
-                continue
-            }
-
-            # If the parent is null or empty then the pom isn't directly including 
-            # one of the pom.[client|data|management].xml and needs to be specially
-            # handled
-            if (("" -eq $parent) -or ($null -eq $parent))
-            {
-                # Cosmos has a root pom which includes pom.client.xml that won't
-                # be detected by this logic. It's easier to deal with specially
-                # than it is to try and climb the pom chain here.
-                if ($Dir.BaseName -eq 'cosmos') 
-                {
-                    $clientArr += $artifactId
-                } 
-                else 
-                {
-                    Write-Host "*snowflake* Pom $PomPath, has a null or empty relative path."
-                }
+    $artifacts = Get-AllPkgProperties -ServiceDirectory $dirName -includeMgmt $false
+    foreach ($pkgInfo in $artifacts) {
+        if (!$serviceName) {
+            $serviceInfo = $metadata | ? { $_.Package -eq $pkgInfo.Name -and $_.GroupId -eq $pkgInfo.Group} 
+            if ($serviceInfo -and $serviceInfo.ServiceName) {
+                $serviceName = $serviceInfo.ServiceName
             } 
-            else 
-            {
-                if (($parent.IndexOf('azure-client-sdk-parent') -ne -1) -or ($parent.IndexOf('azure-data-sdk-parent') -ne -1))
-                {
-                    $clientArr += $artifactId
-                } 
-                else 
-                {
-                    $mgmtArr += $artifactId
-                }
+            else {
+                $serviceName = $dirName.ToUpper()
```

Comment:
Instead of doing a blind ToUpper can we should do CultureInfo.CurrentCulture.TextInfo.ToTitleCase($dirName) so that it works for a lot of service names automatically but only if there is only one word in the service name.